internal/command/scale/count_machines.go

```diff
@@ -69,6 +70,21 @@ func runMachinesScaleCount(ctx context.Context, appName string, appConfig *appco
 		return err
 	}
 
+	// Add env variable overrides to launch configs
+	if env := flag.GetStringArray(ctx, "env"); len(env) > 0 {
+		parsedEnv, err := cmdutil.ParseKVStringsToMap(env)
+		if err != nil {
+			return fmt.Errorf("failed parsing environment: %w", err)
+		}
+		lo.ForEach(actions, func(plan *planItem, _ int) {
+			c := plan.LaunchMachineInput.Config
+			if c.Env == nil {
+				c.Env = make(map[string]string)
+			}
```

Comment:
You can remove these lines. `lo.Assign` does the right thing when `c.Env` is `nil`